### PR TITLE
[rocm7.0_internal_testing] skip 3D NCHW FP16 batchnorm test due to Native accuracy issue

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
Skip for `test_nn.py::TestNN.test_batchnorm_3D_train_NCHW_vs_native_mixed_float16`
Test failed on `weight gradient` comparison MIOpen/CuDNN vs Native batchnorm. 

But CPU test `test_batchnorm_3D_train_NCHW_vs_cpu_mixed_float16` passed.
It looks like FP16 Native batchnorm issue.

Failed on MI200/MI300 and V100
It passed somehow on Navi (with enabled MIOpen)

Fixes SWDEV-541024, SWDEV-539171

```
python test_nn.py -v -k test_batchnorm_3D_train_NCHW_vs_native_mixed_float16

test_batchnorm_3D_train_NCHW_vs_native_mixed_float16 (__main__.TestNN) ... skipped '3D float16 NCHW train failed on CUDA and ROCm due to Native batchnorm accuracy issue SWDEV-541024'

OK (skipped=1)
```

Cherry-picked to release/2.7 branch via https://github.com/ROCm/pytorch/pull/2390

Cherry-picked to release/2.6 branch via https://github.com/ROCm/pytorch/pull/2391

Cherry-picked to release/2.8 branch via https://github.com/ROCm/pytorch/pull/2652

Cherry-picked to release/2.9 branch via https://github.com/ROCm/pytorch/pull/2788

### CODE REVIEW

**Review by pruthvistony — APPROVED** [review]

**Comment by dnikolaev-amd** [comment]
! cherry-pick --onto release/2.7

**Comment by okakarpa** [comment]
Created branch [autogenerated/release/2.7_cherry-pick_pr-2370](https://github.com/ROCm/pytorch/tree/autogenerated/release/2.7_cherry-pick_pr-2370) and https://github.com/ROCm/pytorch/pull/2390

**Comment by okakarpa** [comment]
Created branch [autogenerated/release/2.6_cherry-pick_pr-2370](https://github.com/ROCm/pytorch/tree/autogenerated/release/2.6_cherry-pick_pr-2370) and https://github.com/ROCm/pytorch/pull/2391

**Comment by dnikolaev-amd** [comment]
! cherry-pick --onto release/2.8

**Comment by dhonnappa-amd** [comment]
Created branch [autogenerated/release/2.8_cherry-pick_pr-2370](https://github.com/ROCm/pytorch/tree/autogenerated/release/2.8_cherry-pick_pr-2370) and https://github.com/ROCm/pytorch/pull/2652

Comment processed by [Build](https://ml-ci-internal.amd.com/job/pytorch/job/pytorch-ci-comments-interpreter-pipeline/290901/)

**Comment by dnikolaev-amd** [comment]
! cherry-pick --onto release/2.9
